paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/interceptor/ConfirmationTokenConfirmationInterceptorTest.kt

```diff
@@ -721,6 +725,129 @@ class ConfirmationTokenConfirmationInterceptorTest {
         assertTrue(clientContextParams.requireCvcRecollection!!)
     }
 
+    @OptIn(PaymentMethodOptionsSetupFutureUsagePreview::class)
+    @Test
+    fun `SFU priority - user checkbox takes highest priority over PMO SFU for New payment method`() =
+        runConfirmationTokenInterceptorScenario(
+            initializationMode = PaymentElementLoader.InitializationMode.DeferredIntent(
+                intentConfiguration = PaymentSheet.IntentConfiguration(
+                    mode = PaymentSheet.IntentConfiguration.Mode.Payment(
+                        amount = 1099L,
+                        currency = "usd",
+                        paymentMethodOptions = PaymentSheet.IntentConfiguration.Mode.Payment.PaymentMethodOptions(
+                            mapOf(
+                                PaymentMethod.Type.Card to PaymentSheet.IntentConfiguration.SetupFutureUse.OnSession
+                            )
+                        )
+                    ),
+                )
+            ),
+        ) { interceptor ->
+            val confirmationOption = PaymentMethodConfirmationOption.New(
+                createParams = PaymentMethodCreateParamsFixtures.DEFAULT_CARD,
+                optionsParams = PaymentMethodOptionsParams.Card(
+                    setupFutureUsage = ConfirmPaymentIntentParams.SetupFutureUsage.OffSession
+                ),
+                extraParams = null,
+                shouldSave = true,
+                passiveCaptchaParams = null,
+            )
+
+            val params = interceptor.prepareConfirmationTokenParams(confirmationOption, null)
```

Comment:
Calling this method feels bad, and using a network test would likely provide a much better actual test.